### ISSUE TITLE
[JENKINS-37233] Switch to a responsive table component

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#3097]
<p>This looks pretty excellent and we will want this kind of functionality <a href="https://facebook.github.io/fixed-data-table/example-filter.html" class="external-link" target="_blank" rel="nofollow noopener">https://facebook.github.io/fixed-data-table/example-filter.html</a></p>

<p><b>In Scope</b></p>
<ul>
	<li>Can you please investigate the feasibility of this?</li>
	<li>How well would this work on mobile?</li>
	<li>If it looks good and we don't think it will take a long time lets switch? (have a chat with Michael before starting)</li>
</ul>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay">jamesdumay</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233" target="_blank">Switch to a responsive table component</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers">cliffmeyers</a>
<li><b>status</b>: Resolved
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Postponed
<li><b>resolved</b>: 2016-09-21T22:57:06+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>This looks pretty excellent and we will want this kind of functionality <a href="https://facebook.github.io/fixed-data-table/example-filter.html" class="external-link" target="_blank" rel="nofollow noopener">https://facebook.github.io/fixed-data-table/example-filter.html</a></p>

<p><b>In Scope</b></p>
<ul>
	<li>Can you please investigate the feasibility of this?</li>
	<li>How well would this work on mobile?</li>
	<li>If it looks good and we don't think it will take a long time lets switch? (have a chat with Michael before starting)</li>
</ul>
</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233?focusedId=268961&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-268961">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> would switching help with things like <a href="https://issues.jenkins.io/browse/JENKINS-35996" title="Animate new rows appearing on SSE enabled tables" class="issue-link" data-issue-key="JENKINS-35996"><del>JENKINS-35996</del></a>?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> would switching help with things like <a href="https://issues.jenkins.io/browse/JENKINS-35996" title="Animate new rows appearing on SSE enabled tables" class="issue-link" data-issue-key="JENKINS-35996"><del>JENKINS-35996</del></a>?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233?focusedId=269069&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-269069">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a>  more than likely, yes. The reason the row animation was a problem is that you can't animate the height or max-height of TR elements (and our JDL Table is indeed based on HTML table / tr / td elements). Before we jump into using any particular React data table I think we should do a quick evaluation of the different tables just to make sure we are picking the right one.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a>  more than likely, yes. The reason the row animation was a problem is that you can't animate the height or max-height of TR elements (and our JDL Table is indeed based on HTML table / tr / td elements). Before we jump into using any particular React data table I think we should do a quick evaluation of the different tables just to make sure we are picking the right one.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233?focusedId=269130&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-269130">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> go for it <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>One thing that I like about this is filtering, sorting and mobile responsiveness. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> go for it <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

<p>One thing that I like about this is filtering, sorting and mobile responsiveness. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233?focusedId=269960&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-269960">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a>  the API looks pretty decent, in particular it's crucial that the grid allows custom JSX for each cell so that we can customize it with anything we dream up. I tested it on my Nexus 6P and it seems decent, although some oddities:</p>
<ul>
	<li>Sorting did not work at all; would need to debug to find out why. Maybe it's not listening for the right gesture events.</li>
	<li>Since the grid is virtualized, you can only command-F against the visible data. This will be an issue with any virtualized grid of course.</li>
	<li>Filtering is not really "baked in"; they're just writing a bunch of custom logic and updating the data provider to the grid. This is the same amount of code you'd write for any filtered list.</li>
	<li>Need to see how the dimensions work. They want explicit pixel dimensions for the width and height of the grid. It doesn't seem to support percentages, strangely. That seems... very unresponsive to me. Although we could write a JSX component we use to wrap the grid and specify appropriate width or height based on browser's resize event.</li>
</ul>


<p>I'd say this might be worth spiking a POC on a single screen, perhaps pipelines or activity? Maybe we can touch base on it when you are back next week and I whip something together real quick. I suspect I could adapt one of our screens in a few hours.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay" class="user-hover" rel="jamesdumay">jamesdumay</a>  the API looks pretty decent, in particular it's crucial that the grid allows custom JSX for each cell so that we can customize it with anything we dream up. I tested it on my Nexus 6P and it seems decent, although some oddities:</p>
<ul>
	<li>Sorting did not work at all; would need to debug to find out why. Maybe it's not listening for the right gesture events.</li>
	<li>Since the grid is virtualized, you can only command-F against the visible data. This will be an issue with any virtualized grid of course.</li>
	<li>Filtering is not really "baked in"; they're just writing a bunch of custom logic and updating the data provider to the grid. This is the same amount of code you'd write for any filtered list.</li>
	<li>Need to see how the dimensions work. They want explicit pixel dimensions for the width and height of the grid. It doesn't seem to support percentages, strangely. That seems... very unresponsive to me. Although we could write a JSX component we use to wrap the grid and specify appropriate width or height based on browser's resize event.</li>
</ul>


<p>I'd say this might be worth spiking a POC on a single screen, perhaps pipelines or activity? Maybe we can touch base on it when you are back next week and I whip something together real quick. I suspect I could adapt one of our screens in a few hours.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233?focusedId=269961&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-269961">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The grid does handle horizontal scrolling pretty well, when the grid's width is larger than the viewport.</p>
</pre>
</details>

</ul>
</details>

<p>The grid does handle horizontal scrolling pretty well, when the grid's width is larger than the viewport.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233?focusedId=270808&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-270808">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Some more detail from initial spike on this grid:</p>

<ol>
	<li>The grid requires an explicit height and width. I wrote a wrapper component that listens for resize events and then updates the props on the FDT, and that seems to work well. If the wrapper could be sized using CSS top/bottom constraints then the grid would behave nicely. This would be easy for Activity / Branches / PR's screens but the main Pipelines screens and variable height Favorites / ExtensionPoint area.</li>
	<li>For width - especially on mobile - we could probably allow the grid to exceed the size of the wrapper and leverage its built-in horizontal scroll capability. So the wrapper would probably have a "minGridWidth" property that we could set to maybe something like 800 or 1000px and on mobile the user could scroll left/right. On larger screens the grid would just be stretched to fill the available space.</li>
	<li>Columns also require an explicit width. We could allow the wrapper to define something more flexible - like percentages - and then translate that pixel values that are passed down to the columns.</li>
	<li>There's a desire to hide certain columns on mobile; normally would be done with media queries. I am not sure the grid would respond well to that based on how it measures itself; again we might be able to leverage something in the wrapper that could do some tricks and then turn certain columns on and off.</li>
</ol>

</pre>
</details>

</ul>
</details>

<p>Some more detail from initial spike on this grid:</p>

<ol>
	<li>The grid requires an explicit height and width. I wrote a wrapper component that listens for resize events and then updates the props on the FDT, and that seems to work well. If the wrapper could be sized using CSS top/bottom constraints then the grid would behave nicely. This would be easy for Activity / Branches / PR's screens but the main Pipelines screens and variable height Favorites / ExtensionPoint area.</li>
	<li>For width - especially on mobile - we could probably allow the grid to exceed the size of the wrapper and leverage its built-in horizontal scroll capability. So the wrapper would probably have a "minGridWidth" property that we could set to maybe something like 800 or 1000px and on mobile the user could scroll left/right. On larger screens the grid would just be stretched to fill the available space.</li>
	<li>Columns also require an explicit width. We could allow the wrapper to define something more flexible - like percentages - and then translate that pixel values that are passed down to the columns.</li>
	<li>There's a desire to hide certain columns on mobile; normally would be done with media queries. I am not sure the grid would respond well to that based on how it measures itself; again we might be able to leverage something in the wrapper that could do some tricks and then turn certain columns on and off.</li>
</ol>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-37233?focusedId=270810&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-270810">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>We decided to shelve any further work on this and let <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=brody" class="user-hover" rel="brody">brody</a> instead do a more comprehensive assessment of the app on mobile, which will drive out requirements and lead us to a better solution.</p>

<p>For reference: Brody also dropped this link which contains an interesting example of a single scrollbar that handles scrolling through two different lists of data. Might be useful for how we decide to handle the layout of favorites and pipelines list. <a href="https://codepen.io/jgx/pen/wiIGc" class="external-link" target="_blank" rel="nofollow noopener">https://codepen.io/jgx/pen/wiIGc</a></p>
</pre>
</details>

</ul>
</details>

<p>We decided to shelve any further work on this and let <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=brody" class="user-hover" rel="brody">brody</a> instead do a more comprehensive assessment of the app on mobile, which will drive out requirements and lead us to a better solution.</p>

<p>For reference: Brody also dropped this link which contains an interesting example of a single scrollbar that handles scrolling through two different lists of data. Might be useful for how we decide to handle the layout of favorites and pipelines list. <a href="https://codepen.io/jgx/pen/wiIGc" class="external-link" target="_blank" rel="nofollow noopener">https://codepen.io/jgx/pen/wiIGc</a></p>
